### PR TITLE
Match all of JStage (code borrowed from TWW)

### PR DESCRIPTION
**Pull request by break-core**
This PR matches all of JStage, a library which provides some generic classes like TActor, TObject, TCamera, and more.

I wanted to match this before the end of the year really quickly, as matching JStage is basically required for doing JDrama.

This also adds `FLOAT_MAX` to `dolphin/types.h` (required to match `JStage::TCamera::JSGGetProjectionFar`, but may be useful in the future)

### CODE REVIEW

**Comment by Antidote** [comment]
FLOAT_MAX belongs in the stdc headers not types.h

File: include/JSystem/JStage/JSGActor.hpp [hunk 1]
```diff
@@ -1,30 +1,27 @@
-#ifndef JSG_ACTOR_HPP
-#define JSG_ACTOR_HPP
+#ifndef JSG_ACTOR_H
+#define JSG_ACTOR_H
```
Comment (Mrkol) [inline include/JSystem/JStage/JSGActor.hpp]: Please format include guards as FILE_NAME_BUT_IN_UPPER_SNAKE_CASE, i.e. `JSG_ACTOR_HPP` here

File: include/JSystem/JStage/JSGActor.hpp [hunk 2]
```diff
@@ -1,30 +1,27 @@
...
 
-#include <dolphin/mtx.h>
-#include <JSystem/JStage/JSGObject.hpp>
+#include "JSystem/JStage/JSGObject.hpp"
+#include "dolphin/mtx.h"
```
Comment (Mrkol) [inline include/JSystem/JStage/JSGActor.hpp]: please use <> for absolute-y/system-y includes (i.e. everywhere in jsystem)

File: include/JSystem/JStage/JSGActor.hpp [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
 
 namespace JStage {
+    struct TActor : public TObject {
```
Comment (Mrkol) [inline include/JSystem/JStage/JSGActor.hpp]: Don't indent inside of namespaces, it's waste of horizontal space

File: src/JSystem/JStage/JSGActor.cpp [hunk 4]
```diff
@@ -1 +1,78 @@
-#include <JSystem/JStage/JSGActor.hpp>
+#include "JSystem/JStage/JSGActor.hpp"
+#include "dolphin/types.h"
+
+/* 8026DBBC-8026DC1C       .text __dt__Q26JStage6TActorFv */
```
Comment (Mrkol) [inline src/JSystem/JStage/JSGActor.cpp]: purge these comments, they are not relevant for sms (offsets would be different)

File: src/JSystem/JStage/JSGSystem.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+s32 JStage::TSystem::JSGFGetType() const 
+{
+    return 1;
+}
+
+void* JStage::TSystem::JSGFindObject(const char*, JStage::TEObject) const 
+{
+    return NULL;
```
Comment (Mrkol) [inline src/JSystem/JStage/JSGSystem.cpp]: use nullptr, same as the rest of the codebase (and several other decomp projects)

**Review by Mrkol — CHANGES_REQUESTED** [review]

**Pull request merged**